### ISSUE TITLE
Hangs on "Loading Contact", and native "Send Text" app doesn't show up

### ISSUE THREAD

**Issue by CobreDev** [runnersaw/pebble-text#19]
If I pirate this tweak (I really hate pirating it, but it's the only way I can get it to work, and I can't buy it on BigBoss, so sorry), it works like a charm. It's version 3.0.0, but if I compile 3.0.0 from here, the native Send Text app doesn't show up. I downloaded the "TextSender for Pebble" app from the Rebble store, and it just hangs on "Loading Contact...". I haven't ever been able to get past this on the compiled GitHub version.

**My Device:** iPhone 5s, running unc0ver on 12.1.2.
**My Pebble:** Pebble Time Kickstarter, and Pebble Classic (The classic always gets stuck at the contact screen, even with the pirated version)

**Comment by joemerlino** [comment]
@mac-user669 can you provide your THEOS configuration? SDK and so on?

**Comment by CobreDev** [comment]
@joemerlino I don't know how to use THEOS, so I just made a DEBIAN folder in the "pebblesmstweak" folder, and ran `dpkg -b pebblesmstweak` to create the deb. Is there a better way?

**Comment by joemerlino** [comment]
@mac-user669 yes, configure a THEOS environment to recompile the tweak. 

More info here: https://iphonedevwiki.net/index.php/Theos/Setup

**Comment by CobreDev** [comment]
I haven’t found a good tutorial for doing that, could you please walk me through it?

> On Apr 2, 2019, at 05:58, Giuseppe Merlino <notifications@github.com> wrote:
> 
> @mac-user669 yes, configure a THEOS environment to recompile the tweak.
> 
> More info here: https://iphonedevwiki.net/index.php/Theos/Setup
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.


Edit: Spelling

**Comment by CobreDev** [comment]
@joemerlino Could you guide me through setting up THEOS?

**Comment by CobreDev** [comment]
I've finally installed THEOS, but when I try to make it, it gives me this:
`> Making all for tweak pebblesmstweak…
==> Preprocessing main.xm…

==> Compiling main.xm (armv7)…

main.xm:4:9: fatal error: 'ChatKit/CKComposition.h' file not found
#import <ChatKit/CKComposition.h>
        ^~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[3]: *** [/Users/cooperhull/Downloads/pebble-text-master/pebblesmstweak/.theos/obj/debug/armv7/main.xm.72fe1e4b.o] Error 1
rm /Users/cooperhull/Downloads/pebble-text-master/pebblesmstweak/.theos/obj/debug/armv7/main.xm.mm
make[2]: *** [/Users/cooperhull/Downloads/pebble-text-master/pebblesmstweak/.theos/obj/debug/armv7/pebblesmstweak.dylib] Error 2
make[1]: *** [internal-library-all_] Error 2
make: *** [pebblesmstweak.all.tweak.variables] Error 2`


Any way to fix this?

**Comment by runnersaw** [comment]
Hey there, sorry I'm not very active on this nowadays.

To import headers for private iOS frameworks, you'll have to create a folder and add the header to it in a particular location, please see: https://www.reddit.com/r/jailbreakdevelopers/comments/44e14h/whats_the_best_way_to_import_a_class_from_a/

As far as getting the headers, you'll have to search around for iOS private/runtime headers and find the ones you need to copy into that folder, or class dump them yourself.

**Comment by mac-user669** [comment]
Hmm, I see the words in front of my face, but I don't have much THEOS understanding so I don't have any idea what you're talking about. Could you ELI5, and walk me through doing this? Thanks for replying, I love using your tweak and apps!

**Comment by mac-user669** [comment]
I have THEOS on my Raspberry Pi 3b+, and I think I installed the right headers. However, when I run `make package`, it gives me this error:
https://pastebin.com/7FT7gWB2

I think it's because something with Swift, but I'm not sure how to fix it. Would it be better to install THEOS on a Windows machine?

**Comment by mac-user669** [comment]
Ignore my last comment, I had something wrong. I now have THEOS on my MacBook Air running Mojave, and I'm using the 11.2 sdk. Where should I put the ChatKit header?

**Comment by CobreDev** [comment]
Hey, is there any way you could compile this for me? All I'm getting is errors. Thanks!

**Comment by joemerlino** [comment]
@runnersaw got it compiled, trying to make it compatible with iOS 12.X can you explain the difference between master and production branches? thanks!

**Comment by runnersaw** [comment]
@joemerlino The only differences are that the production branch i) changes the bundle ID, title, and description of the tweak from "Beta" to production and ii) disables logging.

**Comment by joemerlino** [comment]
Gotcha! thanks :)

**Comment by CobreDev** [comment]
I'm thinking I should just close this issue, since a fix is being worked on. Thanks for all of your help everyone!
